include/swift/Runtime/Config.h

```diff
@@ -38,7 +38,7 @@
 #ifndef SWIFT_USE_SWIFTCALL
 // Clang doesn't support mangling functions with the swiftcall attribute
 // on Windows and crashes during compilation: http://bugs.llvm.org/show_bug.cgi?id=32000
-#if (__has_attribute(swiftcall) || defined(__linux__)) && !defined(_WIN32)
+#if __has_attribute(swiftcall) && !defined(_WIN32)
```

Comment:
I can't think of a legitimate reason for any host compiler code to directly include definitions from the runtime. It seems to me it'd be better to factor this out into constants in `swift/ABI` that declare whether each target platform uses the swiftcall convention, like we do for other target-dependent things like pointer masks.